students/chieu_quach/lesson04/mailroom_part2.py

```diff
@@ -2,6 +2,9 @@
 # Author     -  Chieu Quach
 # Assignment -  Lesson 4
 # Exercise   -  Mailroom Part 2
+#               Prints list of users from donation_list.
+#               Update new record if username is not found in list
+
 
 import sys
 global donation_list, len_donation_list
```

Comment:
no need for `global` for `donation_list`

your donation_list:
```
donation_list = {1: {'first name': 'William', 'last name': 'Gates III', 'Amount': '653684.49',
                     'num_gift': '2', 'avg_amt': '326892.24'},
                 2: {'first name': 'Mark', 'last name': 'Zukerberg', 'Amount': '16396.10',
                     'num_gift': '3', 'avg_amt': '5465.37'},
                 3: {'first name': 'Jeff', 'last name': 'Bezos', 'Amount': '877.33',
                     'num_gift': '1', 'avg_amt': '877.33'},
                 4: {'first name': 'Paul', 'last name': 'Allen', 'Amount': '708.42',
                     'num_gift': '3', 'avg_amt': '236.14' },
                            
                }
```


it is better to use donor name as a key so that your look ups are easy and fast, for example:
```
donors = {"Jeff Bezos": [100, 200, 33], "Joe Bidden": [390483, 20200]}
```